Sources/Factory/Factory/ContainerTrait.swift

```diff
@@ -0,0 +1,25 @@
+import Testing
+
+/// ``ContainerTrait`` is a test trait that provides a scoped container for dependency injection in tests.
+/// It allows you to isolate the default ``Container`` to a test case, thus allowing you to run Swift Testing tests in parallel.
+/// If you use a custom container, you have to create your own trait that conforms to the `TestTrait` protocol.
+/// It is also possible to leverage this behavior in `XCTestCase`, by using the `@TaskLocal` provided `withValue` method. See examples in the ``ParallelXCTests`` file.
+public struct ContainerTrait: TestTrait, TestScoping {
```

Comment:
Any way to arrange things so custom containers have less work to do?

Reply:
That’s going to be tough I think because `@
TaskLocal` cannot be required through a protocol. So custom containers will all need to create their own traits.